src/resolve/resolved-manager.c

```diff
@@ -1060,6 +1060,63 @@ static int manager_ipv6_send(
         return sendmsg_loop(fd, &mh, 0);
 }
 
+int send_dns_notification(Manager* m, DnsAnswer* answer, DnsQuestion* question)
+{
+        _cleanup_(dns_resource_record_unrefp) DnsResourceRecord *canonical = NULL;
+        _cleanup_free_ char *normalized = NULL;
+        DnsResourceRecord *rr;
+        int ifindex, r;
+        _cleanup_(json_variant_unrefp) JsonVariant *array = NULL;
+        VarlinkConnection* connection = NULL;
+
+        DNS_ANSWER_FOREACH_IFINDEX(rr, ifindex, answer) {
+                _cleanup_(json_variant_unrefp) JsonVariant *entry = NULL;
+                int family;
+                const void *p;
+
+                if (rr->key->type == DNS_TYPE_A) {
+                        family = AF_INET;
+                        p = &rr->a.in_addr;
+                } else if (rr->key->type == DNS_TYPE_AAAA) {
+                        family = AF_INET6;
+                        p = &rr->aaaa.in6_addr;
+                }
+                else
+                        continue;
```

Comment:
i think it would be better to generically allow tracing of any form of RR lookup, not just A and AAAA. Let clients figure out what they want.

Reply:
@poettering My original thought was to have additional methods in the new varlink interface to monitor other types of lookups, but I see the value in this too.

That said, I'm not familiar with other record types at this time to determine what information should be included in such notifications.

What do you think about making this method extensible for future but not actually sending out notifications for other records with this PR? We could add a "type" field indicating the RR type. It would currently only send A and AAAA types with the hostname/ip address info, but would remain extensible for the future.

Reply:
@poettering My original thought was to have additional methods in the new varlink interface to monitor other types of lookups, but I see the value in this too.

That said, I'm not familiar with other record types at this time to determine what information should be included in such notifications.

What do you think about making this method extensible for future but not actually sending out notifications for other records with this PR? We could add a "type" field indicating the RR type. It would currently only send A and AAAA types with the hostname/ip address info, but would remain extensible for the future.

Reply:
Yes, this sounds very reasonable. A and AAAA would be the most useful. Interested people can add other ones later.

Reply:
Thanks - in the latest patch, I've included a "type" field which will currently be set to "A" or "AAAA"